src/utils/sqs_consumer.ts

```diff
@@ -0,0 +1,89 @@
+import { SQS } from 'aws-sdk';
+
+import { ONE_SECOND_MS } from '../constants';
+import { logger } from '../logger';
+
+import { SqsClient } from './sqs_client';
+
+export class SqsConsumer {
+    private readonly _sqsClient: SqsClient;
+    private readonly _beforeHandle?: () => Promise<boolean>;
+    private readonly _handleMessage: (message: SQS.Types.Message) => Promise<any>;
+    private readonly _afterHandle?: () => Promise<void>;
+    private _isConsuming: boolean;
+
+    constructor(params: {
+        sqsClient: SqsClient;
+        beforeHandle?: () => Promise<boolean>;
+        handleMessage: (message: SQS.Types.Message) => Promise<any>;
```

Comment:
Can we save this function callback as a separate type?

Reply:
sure!